.github/workflows/ipa-tests.yml

```diff
@@ -124,30 +130,111 @@ jobs:
               --pkcs12-password Secret.123
           docker exec ipa pki -n ipa-ca-agent ca-user-show admin
 
-      - name: Gather config files
+      - name: Enable ACME in IPA container
+        run: |
+          docker exec ipa ipa-acme-manage enable
+          docker exec ipa ipa-acme-manage status
+          echo "Available" > expected
+          docker exec ipa bash -c "pki acme-info | sed -n 's/\s*Status:\s\+\(\S\+\).*/\1/p' > ${PKIDIR}/actual"
+          diff expected actual
+
+      - name: Run client container
+        run: |
+          docker run \
+              --detach \
+              --name=client \
+              --hostname=client.example.com \
+              --privileged \
+              --tmpfs /tmp \
+              --tmpfs /run \
+              --volume /sys/fs/cgroup:/sys/fs/cgroup:ro \
+              --interactive \
+              ipa:latest \
+              /usr/sbin/init
+
+      - name: Connect client container to network
+        run: docker network connect example client --alias client.example.com
+
+      - name: Install dependencies in client container
+        run: |
+          docker exec client dnf install -y dnf-plugins-core
+          docker exec client dnf copr enable -y @freeipa/freeipa-master-nightly
+          docker exec client dnf install -y freeipa-client certbot
+
+      - name: Install IPA client in client container
+        run: |
+          docker exec client sysctl net.ipv6.conf.lo.disable_ipv6=0
+          docker exec client ipa-client-install \
+              -U \
+              --server=ipa.example.com \
+              --domain=example.com \
+              --realm=EXAMPLE.COM \
+              -p admin \
+              -w Secret.123
+          docker exec client bash -c "echo Secret.123 | kinit admin"
+          docker exec client klist
+
+      - name: Verify certbot in client container
+        run: |
+          docker exec client certbot register \
+              --server https://ipa.example.com/acme/directory \
```

Comment:
To duplicate what an IPA client would do I'd change the URL to ipa-ca.example.com

Reply:
Doesn't that only work when IPA is providing the DNS? Here Docker is providing the DNS, based on the hostnames given in the `docker network connect` command.

Reply:
I don't know. I suppose this shows whether ACME works at all with IPA and it can be left to IPA CI to determine if it works the rest of the way if IPA-based DNS cannot be used.

Reply:
I added an alias for `ipa-ca.example.com`, we'll see if that works.

Reply:
It seems to be working.